app/src/main/java/org/p2p/wallet/alarmlogger/logger/AlarmErrorsLogger.kt

```diff
@@ -0,0 +1,165 @@
+package org.p2p.wallet.alarmlogger.logger
+
+import timber.log.Timber
+import java.math.BigInteger
+import kotlinx.coroutines.launch
+import org.p2p.core.model.CurrencyMode
+import org.p2p.core.token.Token
+import org.p2p.wallet.BuildConfig
+import org.p2p.wallet.alarmlogger.api.AlarmErrorsServiceApi
+import org.p2p.wallet.alarmlogger.model.AlarmSendErrorConverter
+import org.p2p.wallet.alarmlogger.model.AlarmSwapErrorConverter
+import org.p2p.wallet.alarmlogger.model.SwapAlarmError
+import org.p2p.wallet.bridge.interactor.EthereumInteractor
+import org.p2p.wallet.common.di.AppScope
+import org.p2p.wallet.feerelayer.model.RelayAccount
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.jupiter.statemanager.SwapState
+import org.p2p.wallet.newsend.model.SearchResult
+import org.p2p.wallet.utils.Base58String
+import org.p2p.wallet.utils.retryRequest
+import org.p2p.wallet.utils.toBase58Instance
+
+class AlarmErrorsLogger(
+    private val api: AlarmErrorsServiceApi,
+    private val tokenKeyProvider: TokenKeyProvider,
+    private val swapErrorConverter: AlarmSwapErrorConverter,
+    private val sendErrorConverter: AlarmSendErrorConverter,
+    private val ethereumInteractor: EthereumInteractor,
+    private val appScope: AppScope
+) {
+    private class AlarmErrorsError(override val cause: Throwable) : Throwable(cause)
+
+    private val userPublicKey: Base58String
+        get() = tokenKeyProvider.publicKey.toBase58Instance()
+
+    private val isReleaseBuild: Boolean
```

Comment:
this value is always inverted, maybe keep it `isDebugBuild` and then check `if (isDebugBuild) return`?

Reply:
yep, agree